corgisim/outputs.py

```
@@ -517,27 +559,56 @@ def write_headers_FSAM(cor_type, band_pass):
             FSAMSP_V = 21238
 
     if 'wide' in cor_type:
-        
-        FSAM_H = 6687
-        FSAM_V = 13738
-        FSAMNAME = 'R1C5'
-        FSAMSP_H = 6687
-        FSAMSP_V = 13738
+        if polaxis in ['0', '10']:
+            FSAM_H =30677.2
+            FSAM_V = 2959.5
+            FSAMNAME = 'OPEN'
+            FSAMSP_H = 30677.2
+            FSAMSP_V = 2959.5
+        else:
+            raise ValueError("Polarimetry mode has not been implemented for FSAM")
 
     if ('spec' in cor_type) & ('rotated' not in cor_type):
         if '2' in band_pass:
-            FSAM_H = 11187
-            FSAM_V = 17438
-            FSAMNAME = 'R2C5'
-            FSAMSP_H = 11187
-            FSAMSP_V = 17438
+            if slit == "R6C5":
+                FSAM_H = 11187
+                FSAM_V = 32638
+                FSAMNAME = 'R6C5'
+                FSAMSP_H = 11187
+                FSAMSP_V = 32638
+            elif slit == "R3C1":
+                FSAM_H = 26937
+                FSAM_V = 21238
+                FSAMNAME = 'R3C1'
+                FSAMSP_H = 26937
+                FSAMSP_V = 21238
+            if slit == "None":
+                FSAM_H =30677.2
+                FSAM_V = 2959.5
+                FSAMNAME = 'OPEN'
+                FSAMSP_H = 30677.2
+                FSAMSP_V = 2959.5
```

some thing here. Should we throw an error if an unsupported one is entered?

---

we have set what slit mask is allowed for a specific band in initialization of CorgiOptics. It is like:
```
if '2' in self.bandpass:     
                spec_kw_allowed = {
                    'slit': ['None', 'R6C5', 'R3C1'],
                    'prism': ['None', 'PRISM3', 'PRISM2']}
            #### allowed slit for band3
elif '3' in self.bandpass:
    spec_kw_allowed = {
        'slit': ['None', 'R1C2', 'R3C1'],
        'prism': ['None', 'PRISM3', 'PRISM2']}
```